video/out/gpu_next/context.c

```diff
@@ -38,13 +48,68 @@
 #endif
 
 struct priv {
+#ifdef PL_HAVE_D3D11
+    pl_d3d11 d3d11;
+#endif
 #ifdef PL_HAVE_OPENGL
     pl_opengl opengl;
 #else
     char dummy;
 #endif
 };
 
+static bool d3d11_pl_init(struct vo *vo, struct gpu_ctx *ctx, struct priv *p,
+                          struct ra_ctx_opts *ctx_opts)
+{
+#if !HAVE_D3D11 || !defined(PL_HAVE_D3D11)
+    MP_MSG(ctx, vo->probing ? MSGL_V : MSGL_ERR,
+           "libplacebo was built without D3D11 support.\n");
+    return false;
+#else // HAVE_D3D11 && defined(PL_HAVE_D3D11)
+    bool success = false;
+
+    ID3D11Device   *device    = ra_d3d11_get_device(ctx->ra_ctx->ra);
+    IDXGISwapChain *swapchain = ra_d3d11_ctx_get_swapchain(ctx->ra_ctx);
+    if (!device || !swapchain) {
+        mp_err(ctx->log,
+               "Failed to receive required components from the mpv d3d11 "
+               "context! (device: %s, swap chain: %s)\n",
+               device    ? "OK" : "failed",
+               swapchain ? "OK" : "failed");
+        goto err_out;
+    }
+
+    p->d3d11 = pl_d3d11_create(ctx->pllog, pl_d3d11_params(
+        .device         = device,
```

Comment:
Minor style nit: This should probably be reindented now the other options are gone

Reply:
Ah yes, fixed this one as I finally noticed this LGTM :) .